app/forms/vaccinate_form.rb

```diff
@@ -75,10 +75,12 @@ def save(draft_vaccination_record:)
     draft_vaccination_record.save # rubocop:disable Rails/SaveBang
   end
 
-  def ask_not_taking_medication? = programme.doubles?
+  def ask_not_taking_medication? = programme.doubles? || programme.flu?
 
   def ask_not_pregnant? = programme.td_ipv?
 
+  def ask_asthma_flare_up? = programme.flu?
```

Comment:
```suggestion
  def ask_asthma_flare_up? = programme.flu? && delivery_method == :nasal_spray
```

I would suggest keeping all the logic on whether to ask the question in the single method so it can be re-used elsewhere if necessary.

Reply:
Yeah I wanted to do that but delivery method is defined in the component and these question-specific methods in the vaccinate_form. I've moved those methods into the component now- does that work?

Reply:
Yeah, that looks good!